addons/binding/org.openhab.binding.alarm/src/main/java/org/openhab/binding/alarm/internal/Countdown.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.alarm.internal;
+
+import java.util.Timer;
+import java.util.TimerTask;
+
+/**
+ * Handles a countdown for the different alarm controller delays.
+ *
+ * @author Gerhard Riegler - Initial contribution
+ */
+public class Countdown {
+    private Timer timer;
+
+    /**
+     * Starts a countdown and gives feedback via the callback.
+     */
+    public void start(final int startFrom, CountdownCallback callback) {
+        stop();
+        timer = new Timer();
```

Comment:
Can you reuse the scheduler from the AlarmThingHandler here?

See D.2: https://docs.openhab.org/developers/development/guidelines.html#d-runtime-behavior